core-web/src/main/java/org/silverpeas/core/web/util/servlet/GoTo.java

```diff
@@ -63,47 +64,60 @@ public void doGet(HttpServletRequest req, HttpServletResponse res)
   @Override
   public void doPost(HttpServletRequest req, HttpServletResponse res)
       throws ServletException, IOException {
-
-    String id = getObjectId(req);
-
+    final HttpRequest httpRequest = HttpRequest.decorate(req);
+    final Context context = new Context(httpRequest, res).init();
+    String id = getObjectId(httpRequest);
     try {
-
-
-      String redirect = getDestination(id, req, res);
+      String redirect = getDestination(id, httpRequest, res);
       if (!StringUtil.isDefined(redirect)) {
-        objectNotFound(req, res);
+        objectNotFound(context);
       } else {
         if (!res.isCommitted()) {
           // The response was not previously sent
-          if (redirect == null || !redirect.startsWith("http")) {
-            redirect = URLUtil.getFullApplicationURL(req) + "/autoRedirect.jsp?" + redirect;
+          if (!redirect.startsWith("http")) {
+            redirect = URLUtil.getFullApplicationURL(httpRequest) + "/autoRedirect.jsp?" + redirect;
           }
-          res.sendRedirect(res.encodeRedirectURL(redirect));
+          res.sendRedirect(res.encodeRedirectURL(UriBuilder.fromUri(redirect)
+              .queryParam("fromResponsiveNavigation", context.isFromResponsiveNavigation()).build()
+              .toString()));
         }
       }
     } catch (AccessForbiddenException afe) {
-      accessForbidden(req, res);
+      accessForbidden(context);
     } catch (Exception e) {
-      objectNotFound(req, res);
+      objectNotFound(context);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "objectNotFound". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1989)